### ISSUE TITLE
You might want to fill the addr struct passed to accept

### ISSUE THREAD

**Issue by junxzm1990** [zardus/preeny#12]
The standard accept function will fill the "addr struct" (passed as argument) with the address of the peer socket. This is not done by the "accept" in desock.c. Some applications will not continue executing until the "addr struct" is filled. For instance, the latest version of nginx will check the address of the peer socket (I tried it with select mode but not sure about other modes).

**Comment by zardus** [comment]
Good catch. Do you have a fixed implementation you could PR, by any chance?

**Comment by junxzm1990** [comment]
I think you can fill the "addr struct" with a fake address (e.g. INADDR_ANY)  and set the "socklen_t * addrlen" correspondingly. That's what I did.

**Comment by zardus** [comment]
Could you push your code as a PR?

**Comment by junxzm1990** [comment]
I am still "borrowing" the preeny code. When all set, I will PR.

**Comment by zardus** [comment]
Awesome, thanks :-)

**Comment by gsingh93** [comment]
7 months and still no PR :p 

Running into the same issue with nginx.

**Comment by gsingh93** [comment]
There are some other interesting issues that might be worth looking into. While I'm having the same issue with addr not being set, I'm finding that eventually it does get set after about a minute... I'm not sure if something in the code is causing it to fall back to original_accept, but I'll try to investigate more.
